packages/nextjs/src/server/protect.ts

```diff
@@ -112,10 +112,8 @@ export const createProtect = (opts: {
 
 const isServerActionRequest = (req: Request) => {
   return (
-    !!req.headers.get(nextConstants.Headers.NextUrl) &&
```

Comment:
Next-url as a header has been dropped from request to server actions since `14.1.0` way before `14.2.2` where our page detection started to fail.